components/brave_shields/browser/ad_block_base_service.cc

```diff
@@ -141,6 +141,30 @@ void AdBlockBaseService::ShouldStartRequest(
   //  << ", url.spec(): " << url.spec();
 }
 
+base::Optional<std::string> AdBlockBaseService::GetCspDirectives(
+    const GURL& url,
+    blink::mojom::ResourceType resource_type,
+    const std::string& tab_host) {
+  DCHECK(GetTaskRunner()->RunsTasksInCurrentSequence());
+
+  // Determine third-party here so the library doesn't need to figure it out.
+  // CreateFromNormalizedTuple is needed because SameDomainOrHost needs
+  // a URL or origin and not a string to a host name.
+  bool is_third_party = !SameDomainOrHost(
+      url,
+      url::Origin::CreateFromNormalizedTuple("https", tab_host.c_str(), 80),
```

Comment:
1) you should std::move(tab_host), as described in the `CreateFromNormalizedTuple` description

2) why not pass the full URL or Origin there from the very beginning?

Reply:
I'm just using the same implementation as `ShouldStartRequest` directly above; would be good to keep the same behavior between the two since I use both the same way in `adblock-rust`. I think it's worth addressing in a followup.

Reply:
yeah, ok - could you file a follow-up to change `tab_host` from `string` to `GURL` in `ShouldStartRequest` ?

Reply:
Filed https://github.com/brave/brave-browser/issues/15291